manifests/database.pp

```diff
@@ -0,0 +1,33 @@
+# sets up separate postgresql instance for pulpcore
+# @api private
+class pulpcore::database {
+
+  class { 'postgresql::globals':
+    version              => '10',
+    client_package_name  => 'rh-postgresql10-postgresql-syspaths',
+    server_package_name  => 'rh-postgresql10-postgresql-server-syspaths',
+    contrib_package_name => 'rh-postgresql10-postgresql-contrib-syspaths',
+    service_name         => 'postgresql',
+    datadir              => '/var/lib/pgsql/data',
+    confdir              => '/var/lib/pgsql/data',
+    bindir               => '/usr/bin',
+  }
+
+  include postgresql::server
+  postgresql::server::db { $pulpcore::database_name:
+    user     => $pulpcore::pulp_user,
+    password => postgresql_password($pulpcore::pulp_user, 'secret'),
+  }
+
+  exec { 'django-admin migrate --noinput':
```

Comment:
This needs an explicit `require => Postgresql::Server::Db[$pulpcore::database_name]` because it looks like it's now ran before the postgres server is up.